### PR TITLE
Updates for the latest book theme

### PR DESCRIPTION
**Pull request by choldgraf**
The latest book theme makes it a bit easier for us to re-use it for our own purposes. This PR adds the `main` branch as a dependency and updates our CSS / HTML as needed.

Main changes:

- We now use `sections/header.html` to add the header, rather than overloading the `layout.html` template
- Updated our CSS since we no longer need to hard-code the y-position of the main content of the page
- Refactored the SCSS to follow more standard layout
- Applied `prettier`

This was triggered because I realized we have a bug on mobile:

**Old**:

chrome_hBoAPUW2FQ

**New**:

chrome_9Wm6OlSBQA

### CODE REVIEW

**Comment by choldgraf** [comment]
going to merge this in since it's squashing a kinda nasty bug, and we can iterate in the future!

File: pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ urls = { Organization = "https://2i2c.org" }
 
 requires-python = ">=3.8"
 dependencies = [
-  "sphinx-book-theme>=0.1.7",
+  "sphinx-book-theme@https://github.com/executablebooks/sphinx-book-theme/archive/refs/heads/master.zip",
```
Comment (damianavila) [inline pyproject.toml:17]: I am usually against these "running" dependencies... we should be working with released packages IMHO (unless you have a **really** good excuse to not do it).

File: src/sphinx_2i2c_theme/assets/styles/sphinx-2i2c-theme.scss [hunk 2]
```diff
@@ -1,123 +1,4 @@
...
-  padding-left: .5em;
-}
-
-/* Topbar of main theme */
-.topbar {
-  z-index: 800;
-}
+@import "abstract/variables";
+@import "sections/header";
+@import "sections/sidebar-primary";
```
Comment (damianavila) [inline src/sphinx_2i2c_theme/assets/styles/sphinx-2i2c-theme.scss:4]: I really like this pattern of importing specific (defined per component) scss files!

**Comment by damianavila** [comment]
@choldgraf, left some post-merge comments for you as feedback 😜

**Pull request merged**